main.py

```diff
@@ -0,0 +1,76 @@
+import discord
+import os
+from discord.ext import commands
+from dotenv import load_dotenv
+from emotes import *
+
+load_dotenv()
+
+intents = discord.Intents.default()
+intents.message_content = True
+
+bot = commands.Bot(command_prefix="!", intents=intents)
+
+@bot.event
+async def on_message(message):
+    if not message.author.bot and message.content.startswith("!actioned"):
+        contributor = get_contributor_by_uid(str(message.author.id))
+        if contributor:
+            contributor.actioned = True
+            await message.channel.send("Thanks for the response! You won't receive any more reminders.")
+            return  # Stop processing other checks if actioned command is received
+
+    for emoji_id, contributor in emoji_id_mapping.items():
+        if emoji_id in message.content:
+            print(f'Emoji Found!', {emoji_id})
+            await message.add_reaction(emoji_id)
+
+            if not contributor.actioned:
+                print(f'Messaging the user, {contributor.uid}')
+                message_link = message.jump_url  # Get the jump URL of the message
+                await send_dm_periodically(bot, contributor, message_link)
```

Comment:
Why do you await this?
Won't this block?

Reply:
> Why do you await this? Won't this block?

I used await as this is a continuous task there are no blocks I can identify by my testing.
However after reviewing things I think I have found a potential block issue outside of this that needs to be checked